pkg/functions/kubeless/driver.go

```diff
@@ -120,17 +140,75 @@ func (d *kubelessDriver) Create(ctx context.Context, f *functions.Function) erro
 	span, ctx := trace.Trace(ctx, "")
 	defer span.Finish()
 
+	funcLimits := &functions.FunctionResources{
+		CPU:    d.funcDefaultLimits.CPU,
+		Memory: d.funcDefaultLimits.Memory,
+	}
+	if f.ResourceLimits.CPU != "" {
+		funcLimits.CPU = f.ResourceLimits.CPU
+	}
+	if f.ResourceLimits.Memory != "" {
+		funcLimits.Memory = f.ResourceLimits.Memory
+	}
+
+	funcRequests := &functions.FunctionResources{
+		CPU:    d.funcDefaultRequests.CPU,
+		Memory: d.funcDefaultRequests.Memory,
+	}
+	if f.ResourceRequests.CPU != "" {
+		funcRequests.CPU = f.ResourceRequests.CPU
+	}
+	if f.ResourceRequests.Memory != "" {
+		funcRequests.Memory = f.ResourceRequests.Memory
+	}
+
+	resourceRequirements := kapi.ResourceRequirements{
+		Limits:   kapi.ResourceList{},
+		Requests: kapi.ResourceList{},
+	}
+	if funcLimits.CPU != "" {
+		qty, err := resource.ParseQuantity(funcLimits.CPU)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing cpu limit '%s'", funcLimits.CPU)
+		}
+		resourceRequirements.Limits[kapi.ResourceCPU] = qty
+	}
+	if funcLimits.Memory != "" {
+		qty, err := resource.ParseQuantity(funcLimits.Memory)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing memory limit '%s'", funcLimits.Memory)
+		}
+		resourceRequirements.Limits[kapi.ResourceMemory] = qty
+	}
+	if funcRequests.CPU != "" {
+		qty, err := resource.ParseQuantity(funcRequests.CPU)
+		if err != nil {
+			return errors.Wrapf(err, "error parsing cpu request '%s'", funcRequests.CPU)
+		}
+		resourceRequirements.Requests[kapi.ResourceCPU] = qty
+	}
+	if funcRequests.Memory != "" {
```

Comment:
Seeing this chunk of code that is essentially the same as the code in NewRiffTalk makes me think that this really needs it's own object to fully handle populating the kapi.ResourceRequirements object. or at the very least this functionality should be moved into the FunctionResources object and then shared here and in the NewRiffTalk method.

Reply:
Sharing code between NewRiffTalk and the Kubeless driver is not really feasible since we currently vendor the riff library with its own version of k8s dependencies. This leads to conflicts when passing around k8s objects, so the resource requirement has to be parsed into the correct version of `kapi.Quantity` in riffTalk.